client/asset/driver.go

```diff
@@ -136,8 +136,8 @@ func OpenWallet(assetID uint32, cfg *WalletConfig, logger dex.Logger, net dex.Ne
 // asset with a corresponding driver registered with this package. For now,
 // token id decoding is deferred to the parent asset.
 func DecodeCoinID(assetID uint32, coinID []byte) (cid string, err error) {
```

Comment:
DecodeCoinID is used everywhere, probably even from within some of the creator functions.  I experienced a deadlock and I think it was this.  Assets() was waiting on the lock and part of the stack trace was missing.